examples/freertos_linux/Linux_gcc_gcp_iot/README.md

```diff
@@ -6,22 +6,41 @@ This example uses the Google Cloud IoT Device SDK for Embedded C to connect a Fr
 
 Follow the steps below to connect the FreeRTOS application to the MQTT bridge.
 
-Before you begin, generate a [public/private key pair](https://cloud.google.com/iot/docs/how-tos/credentials/keys), store the private key in the `examples/freertos_linux/Linux_gcc_gcp_iot` directory, and name the key `ec_private.pem`. 
+1. Go to the repository's root directory.
 
-1. Run `make PRESET=FREERTOS_POSIX_REL` in the root directory of the repository. This command downloads of the [FreeRTOS kernel](https://www.freertos.org/index.html), downloads the [FreeRTOS Linux simulator](https://www.freertos.org/FreeRTOS-simulator-for-Linux.html), and ports the Device SDK to the FreeRTOS application.
+```
+cd iot-device-sdk-embedded-c
+```
+
+1. Generate a [public/private key pair](https://cloud.google.com/iot/docs/how-tos/credentials/keys), and store it in the example's directory.
+
+```
+openssl ecparam -genkey -name prime256v1 -noout -out examples/freertos_linux/Linux_gcc_gcp_iot/ec_private.pem
+openssl ec -in examples/freertos_linux/Linux_gcc_gcp_iot/ec_private.pem -pubout -out examples/freertos_linux/Linux_gcc_gcp_iot/ec_public.pem
+```
 
-2. From the root directory, generate the `Linux_gcc_gcp_iot` application for Linux.
+1. Download the [FreeRTOS kernel](https://www.freertos.org/index.html), the [FreeRTOS Linux simulator](https://www.freertos.org/FreeRTOS-simulator-for-Linux.html), and port the Device SDK to the FreeRTOS application.
 
 ```
-cd examples/freertos_linux/Linux_gcc_gcp_iot \
-make
+make clean_all
```

Comment:
clean_all, too drastic, although it is necessary. Since we don't know what was the previous mbedtls build: 32bit or 64bit. Still I'd give more automagical experience than forcing clean_all always.

Reply:
I agree, that it is too drastic, however this is the Makefile for the examples. 

We can assume that people are building the examples when they check out the project and start evaluating. There, it's very bad if they hit build errors.

The degraded UX is when someone already has a build, then builds the example, then their code doesn't build any more, because the wrong mbedTLS library is there. However this path is less likely at this point of the project.

Nevertheless, in another PR I can fix the mbedTLS library output, so the time window that this risk is there would be small.

WDYT?

Reply:
Unfortunately this means that if I build the library with a specific CONFIG first, like all of our documentation tells the user to do, and THEN build the examples, that the special config is wiped out and replaced with a library built with the default config.

I would rather solve this issue another way. Create new library target path for the 32bit and one for 64-bit libs of mbedTLS, and have the example depend on these directly, building them if necessary or at least alerting the user to their absence.